ddtrace/_trace/utils_botocore/span_pointers.py

```diff
@@ -22,9 +35,148 @@ def extract_span_pointers_from_successful_botocore_response(
+
+    except Exception as e:
+        log.warning(
+            "failed to generate DynamoDB.PutItem span pointer: %s",
+            str(e),
+        )
+        return []
+
+
+def _aws_dynamodb_item_primary_key_from_item(
+    primary_key_field_names: Set[_DynamoDBItemFieldName],
+    item: _DynamoDBItem,
+) -> _DynamoDBItemPrimaryKey:
+    if len(primary_key_field_names) not in (1, 2):
+        raise ValueError(f"unexpected number of primary key fields: {len(primary_key_field_names)}")
+
+    return {
+        primary_key_field_name: _aws_dynamodb_extract_and_verify_primary_key_field_value_item(
+            item, primary_key_field_name
+        )
+        for primary_key_field_name in primary_key_field_names
+    }
+
+
+def _aws_dynamodb_item_span_pointer_description(
+    pointer_direction: _SpanPointerDirection,
+    table_name: _DynamoDBTableName,
+    primary_key: _DynamoDBItemPrimaryKey,
+) -> _SpanPointerDescription:
+    return _SpanPointerDescription(
+        pointer_kind="aws.dynamodb.item",
+        pointer_direction=pointer_direction,
+        pointer_hash=_aws_dynamodb_item_span_pointer_hash(table_name, primary_key),
+        extra_attributes={},
+    )
+
+
+def _aws_dynamodb_extract_and_verify_primary_key_field_value_item(
+    item: _DynamoDBItem,
+    primary_key_field_name: _DynamoDBItemFieldName,
+) -> _DynamoDBItemPrimaryKeyValue:
+    if primary_key_field_name not in item:
+        raise ValueError(f"missing primary key field: {primary_key_field_name}")
+
+    value_object = item[primary_key_field_name]
+
+    if len(value_object) != 1:
+        raise ValueError(f"primary key field {primary_key_field_name} must have exactly one value: {len(value_object)}")
+
+    value_type, value_data = next(iter(value_object.items()))
+    if value_type not in ("S", "N", "B"):
+        raise ValueError(f"unexpected primary key field {primary_key_field_name} value type: {value_type}")
+
+    if not isinstance(value_data, str):
+        raise ValueError(f"unexpected primary key field {primary_key_field_name} value data type: {type(value_data)}")
+
+    return {value_type: value_data}
+
+
+def _aws_dynamodb_item_span_pointer_hash(table_name: _DynamoDBTableName, primary_key: _DynamoDBItemPrimaryKey) -> str:
+    if len(primary_key) == 1:
+        key, value_object = next(iter(primary_key.items()))
+        encoded_key_1 = key.encode("utf-8")
+        encoded_value_1 = _aws_dynamodb_item_encode_primary_key_value(value_object)
+        encoded_key_2 = b""
+        encoded_value_2 = b""
+
+    elif len(primary_key) == 2:
+        (key_1, value_object_1), (key_2, value_object_2) = sorted(
+            primary_key.items(), key=lambda x: x[0].encode("utf-8")
+        )
+        encoded_key_1 = key_1.encode("utf-8")
+        encoded_value_1 = _aws_dynamodb_item_encode_primary_key_value(value_object_1)
+        encoded_key_2 = key_2.encode("utf-8")
+        encoded_value_2 = _aws_dynamodb_item_encode_primary_key_value(value_object_2)
+
+    else:
```

Comment:
```suggestion
    if len(primary_key) not in (1,2):
         raise ValueError(f"unexpected number of primary key fields: {len(primary_key)}")
```

Reply:
nit suggestion to match previous method (and have at the beginning)

Reply:
ok, though i might leave an else statement in case we change the separate initial if statement but somehow forget to fix the rest of the code

Reply:
so it would end up looking like this:

```
    if len(primary_key) not in (1, 2):
        raise ValueError(f"unexpected number of primary key fields: {len(primary_key)}")

    if len(primary_key) == 1:
        key, value_object = next(iter(primary_key.items()))
        encoded_key_1 = key.encode("utf-8")
        encoded_value_1 = _aws_dynamodb_item_encode_primary_key_value(value_object)
        encoded_key_2 = b""
        encoded_value_2 = b""

    elif len(primary_key) == 2:
        (key_1, value_object_1), (key_2, value_object_2) = sorted(
            primary_key.items(), key=lambda x: x[0].encode("utf-8")
        )
        encoded_key_1 = key_1.encode("utf-8")
        encoded_value_1 = _aws_dynamodb_item_encode_primary_key_value(value_object_1)
        encoded_key_2 = key_2.encode("utf-8")
        encoded_value_2 = _aws_dynamodb_item_encode_primary_key_value(value_object_2)

    else:
        # this should never happen
        raise Exception("logical error??")
```

i'm not really a fan of that, but we can do it if y'all think this is better!

Reply:
per slack discussion, we're going to leave the code as it is for now.

Reply:
Good to leave it as-is